File: 프로그래머스/unrated/181905. 문자열 뒤집기/문자열 뒤집기.js
function solution(my_string, s, e) {
    let answer = '';
    let count = 0;
    
    for (let i = 0; i < my_string.length; i++) {
        if (i >= s && i <= e) {
            answer += my_string[e - count];
            count++;
        } else {
            answer += my_string[i];
        }
    }
    
    return answer;
}

File: 프로그래머스/unrated/181905. 문자열 뒤집기/문자열 뒤집기.java
import java.util.Stack;

class Solution {
    public String solution(String my_string, int s, int e) {
        String answer = "";
        Stack<Character> stack = new Stack<>();
        
        for (int i = s; i <= e; i++) {
            stack.push(my_string.charAt(i));
        }
        
        for (int i = 0; i < my_string.length(); i++) {
            if (i >= s && i <= e) {
                answer += stack.pop();
            } else {
                answer += my_string.charAt(i);
            }
        }
        
        return answer;
    }
}